Assets/Scripts/Enemies/BaseEnemyAI.cs

```diff
@@ -0,0 +1,136 @@
+
+    PlayerMovementController player;
+    PlayerCombat playerCombat;
+
+    Vector2 velocity;
+    bool isFacingRight = true;
+    float horizontalMove = 0.0f;
+    float maxSpeedChange;
+    float nextMeleeTime = 0f;
+    float nextProjectileTime = 0f;
+
+    // Values used for animations:
+
+    [SerializeField] bool isFiringProjectile = false;
+    [SerializeField] bool isAttacking = false;
+
+    void FindPlayer()
+    {
+        if (player == null)
+        {
+            player = GameObject.FindWithTag(Constants.PLAYER_TAG).GetComponent<PlayerMovementController>();
+        }
+        if (playerCombat == null)
+        {
+            playerCombat = GameObject.FindWithTag(Constants.PLAYER_TAG).GetComponent<PlayerCombat>();
+        }
+    }
+
+    void Update()
+    {
+        FindPlayer();
+        float distanceToPlayer = Mathf.Abs(player.transform.position.x - transform.position.x);
+        if (distanceToPlayer < detectionRadius && distanceToPlayer > meleeRange && !isAttacking && !isFiringProjectile)
+        {
+            Debug.Log("approaching player");
+            horizontalMove = (player.transform.position - transform.position).normalized.x;
+            if (horizontalMove > 0 && !isFacingRight)
+            {
+                Flip();
+            }
+            else if (horizontalMove < 0 && isFacingRight)
+            {
+                Flip();
+            }
+            if (Time.time >= nextProjectileTime)
+            {
+                Debug.Log("fire");
+                nextProjectileTime = Time.time + projectileAttackData.duration;
+                animator.SetTrigger("ProjectileAttack");
+            }
+        }
+        else
+        {
+            if (distanceToPlayer < meleeRange && Time.time >= nextMeleeTime && !isFiringProjectile)
+            {
+                Debug.Log("attack");
+                nextMeleeTime = Time.time + meleeAttackData.duration;
+                animator.SetTrigger("MeleeAttack");
+            }
+        }
+    }
+
+    void FixedUpdate()
+    {
+        if (isAttacking || isFiringProjectile)
+        {
+            horizontalMove = 0f;
+        }
+        Move(horizontalMove * movement.maxSpeed * Time.fixedDeltaTime);
+    }
+
+    void Move(float move)
+    {
+        velocity = rb.velocity;
+        maxSpeedChange = movement.maxAcceleration * Time.fixedDeltaTime;
+		Vector3 desiredVelocity = new Vector2(move * 10f, rb.velocity.y);
+        velocity.x = Mathf.MoveTowards(velocity.x, desiredVelocity.x, maxSpeedChange);
+        rb.velocity = velocity;
+    }
+
+    private void Flip()
+	{
+		isFacingRight = !isFacingRight;
+		Vector3 localScale = transform.localScale;
+		localScale.x *= -1;
+		transform.localScale = localScale;
+	}
+
+    public void TakeDamage(AttackData attackData, Vector3 attackOrigin)
+    {
+        health.SubtractResource(attackData.damage);
+
+        if (health.GetCurrentValue() <= 0)
+        {
+            Die();
+        }
+    }
```

Comment:
For a more robust solution (non-game jam) would normally highly recommend using an interface like `IDamageable` or something like that. We have defined `TakeDamage` in a number of different places now, and having a unified interface would make it much easier to manage.

For a game jam setting I think it's completely fine to leave as-is for now, though.

Reply:
I think it wouldn't be too much effort to add this, I can look into it for a later update